README.md

```diff
@@ -1,14 +1,615 @@
+
+```kotlin
+val subscription = chat.connection.onStatusChange { statusChange: ConnectionStatusChange ->
+    println(statusChange.toString())
+}
+```
+
+To stop listening to changes, `unsubscribe` method on returned subscription instance:
+
+```kotlin
+subscription.unsubscribe()
+```
+
+## Chat rooms
+
+### Creating or retrieving a chat room
+
+You can create or retrieve a chat room with name `"basketball-stream"` this way:
+
+```kotlin
+val room = chat.rooms.get("basketball-stream", RoomOptions(reactions = RoomReactionsOptions()))
+```
+
+The second argument to `rooms.get` is a `RoomOptions` argument, which tells the Chat SDK what features you would like your room to use and
+how they should be configured.
+
+For example, you can set the timeout between keystrokes for typing events as part of the room options. Sensible defaults for each of the
+features are provided for your convenience:
+
+- A typing timeout (time of inactivity before typing stops) of 5 seconds.
+- Entry into, and subscription to, presence.
+
+The defaults options for each feature may be
+viewed [here](https://github.com/ably/ably-chat-js/blob/main/chat-android/src/main/java/com/ably/chat/RoomOptions.kt).
+
+In order to use the same room but with different options, you must first `release` the room before requesting an instance with the changed
+options (see below for more information on releasing rooms).
+
+Note that:
+
+- If a `release` call is currently in progress for the room (see below), then a call to `get` will wait for that to resolve before resolving
+  itself.
+- If a `get` call is currently in progress for the room and `release` is called, the `get` call will reject.
+
+### Attaching to a room
+
+To start receiving events on a room, it must first be attached. This can be done using the `attach` method.
+
+```kotlin
+// Add a listener so it's ready at attach time (see below for more information on listeners)
+room.messages.subscribe { message: Message ->
+    println(message.toString())
+}
+
+room.attach()
+```
+
+### Detaching from a room
+
+To stop receiving events on a room, it must be detached, which can be achieved by using the `detach` method.
+
+```kotlin
+room.detach()
+```
+
+Note: This does not remove any event listeners you have registered and they will begin to receive events again in the
+event that the room is re-attached.
+
+### Releasing a room
+
+Depending on your application, you may have multiple rooms that come and go over time (e.g. if you are running 1:1 support chat). When you
+are completely finished with a room, you may `release` it which allows the underlying resources to be collected.
+
+```kotlin
+rooms.release("basketball-stream")
+```
+
+Once `release` is called, the room will become unusable and you will need to get a new instance using `rooms.get` should you wish to
+re-start the room.
+
+Note that releasing a room may be optional for many applications.
```

Comment:
As we're going to mention not requiring release, perhaps worth mentioning that the server will tidy up connections etc after a period of time?

Reply:
done